### ISSUE TITLE
Langage service reports an error when Importing a non-exported class in a standalone component. 

### ISSUE THREAD

**Issue by JeanMeche** [angular/angular#49874]
### Which @angular/* package(s) are the source of the bug?

compiler-cli, language-service

### Is this a regression?

No

### Description

```
@Directive({ selector: 'test', standalone: true })
class Dir {}

@Component({
  selector: 'app-root',
  template: '',
  standalone: true,
  imports: [Dir],
})
export class AppComponent {}
```
With this example, the language service  throws `Component imports must be standalone components, directives, pipes, or must be NgModules.`.

I found the error a bit misleading and even more since the code works fine at runtime. 

I've read both #42184 and #42207. 

Am I correct to assume there isn't so much we can do here because of `compileNonExportedClasses : false` 

Or do you have an idea to improve the DX here ?

**Comment by atscott** [comment]
> Am I correct to assume there isn't so much we can do here because of compileNonExportedClasses : false

That's correct. I'll leave this open as a potential for having better recognition that the class isn't exported and potentially producing a different/better error message.

**Comment by robmv** [comment]
I have an issue related to this, a small service that shows a simple overlay component, both defined on the same source file, Since only the service is exposed, it doesn't make sense to export the component. The problem arises at the language server when used from VS Code, all template type checking is gone, not a single error is shown, even if I make an intentional buggy template using nonexistent properties of the component. Compilation from CLI is not affected, at least from a clean build, using --watch sometimes the behaviour of type checking or not the template in the CLI is cached if the export is added/removed.

Reading this issue an the linked ones, look like the solution will be to have better error messages, and that the component needs to be exported, even when using only locally. Please take into account in the error message, not only when the component is imported into another one, but this case when the template is not type checked.

**Comment by atscott** [comment]
Closing as resolved by https://github.com/angular/angular/pull/68454. It missed the feature freeze for v22 but will go out with the v22.1 release of language service. Non exported standalone classes will be compiled.
